src/main/clojure/starfish/core.cljc

```diff
@@ -261,12 +261,12 @@
 ;; Account
 
 (defn remote-account
-  [^String username ^String password]
-  ;; TODO: shouldn't this 
-  (let [^String id (or username (error "Username required!"))
-        ^java.util.Map creds {"username" username
-                         "password" password}]
-    (RemoteAccount/create id creds)))
+  ([token](RemoteAccount/create (Utils/createRandomHexString 32)
```

Comment:
I think we should allow passing an arbitrary credential map here? Perhaps as well as a token option

Reply:
@mikera did you mean having accepting both
- token (as a String)
- credential map (as a map).

Something like this doesn't compile due to `Can't have 2 overloads with same arity`
```clj
(defn remote-account
  ([^String token] (RemoteAccount/create (Utils/createRandomHexString 32)
                                {"token" token}))
  ([^clojure.lang.APersistentMap credentials] (RemoteAccount/create (Utils/createRandomHexString 32)
                                credentials))
  ([^String username ^String password]
   (RemoteAccount/create (Utils/createRandomHexString 32)
                         {"username" username
                          "password" password})))
```